ssl/ssl_lib.c

```diff
@@ -7478,6 +7475,23 @@ int SSL_is_connection(SSL *s)
     return SSL_get0_connection(s) == s;
 }
 
+SSL *SSL_get0_listener(SSL *s)
+{
+#ifndef OPENSSL_NO_QUIC
+    if (!IS_QUIC(s))
+        return s;
```

Comment:
why returning `s` at line 7482 is correct? The reason I ask is that for non-QUIC case we just assume every ssl object is listener. I'm not sure it's desired.